main.tf

```diff
@@ -0,0 +1,666 @@
+resource "aws_cloudwatch_log_group" "CloudGuardAWPSnapshotsUtilsLogGroup" {
+  name              = "/aws/lambda/${aws_lambda_function.CloudGuardAWPSnapshotsUtilsFunction.function_name}"
+  retention_in_days = 30
+  depends_on = [
+    aws_lambda_function.CloudGuardAWPSnapshotsUtilsFunction
+  ]
+  tags = local.common_tags
+}
+
+# AWP proxy lambda function role
+# The CloudGuardAWPSnapshotsUtilsLambdaExecutionRole resource defines an IAM role that is used to allow the CloudGuardAWPSnapshotsUtilsFunction to execute.
+resource "aws_iam_role" "CloudGuardAWPSnapshotsUtilsLambdaExecutionRole" {
+  name        = "CloudGuardAWPLambdaExecutionRole"
+  description = "CloudGuard AWP proxy lambda function execution role"
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Principal = {
+          Service = "lambda.amazonaws.com"
+        }
+        Action = "sts:AssumeRole"
+      }
+    ]
+  })
+  tags = local.common_tags
+}
+
+# The CloudGuardAWPSnapshotsPolicy resource defines an IAM policy that is used to define the permissions for the CloudGuardAWPSnapshotsUtilsFunction.
+resource "aws_iam_policy" "CloudGuardAWPSnapshotsPolicy" {
+  name        = "CloudGuardAWPSnapshotsPolicy"
+  description = "Policy for managing snapshots at client side and delete AWP keys"
+  tags = local.common_tags
+
+  policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Action = [
+          "ec2:CreateTags",
+          "ec2:CopySnapshot",
+          "ec2:CreateSnapshot",
+          "ec2:CreateSnapshots",
+          "ec2:DescribeSnapshots",
+          "ec2:DescribeRegions"
+        ]
+        Resource = "*"
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "ec2:DeleteSnapshot"
+        ]
+        Resource = "*"
+        Condition = {
+          StringEquals = {
+            "aws:ResourceTag/Owner" = "CG.AWP"
+          }
+        }
+      },
+      {
+        Effect = "Allow"
+        Action = [
+          "logs:CreateLogStream",
+          "logs:PutLogEvents"
+        ]
+        Resource = ["${aws_cloudwatch_log_group.CloudGuardAWPSnapshotsUtilsLogGroup.arn}:*"]
+      }
+    ]
+  })
+}
+
+# The CloudGuardAWPSnapshotsUtilsLambdaExecutionRoleAttachment resource attaches the CloudGuardAWPSnapshotsPolicy to the CloudGuardAWPSnapshotsUtilsLambdaExecutionRole.
+resource "aws_iam_role_policy_attachment" "CloudGuardAWPSnapshotsUtilsLambdaExecutionRoleAttachment" {
+  role       = aws_iam_role.CloudGuardAWPSnapshotsUtilsLambdaExecutionRole.name
+  policy_arn = aws_iam_policy.CloudGuardAWPSnapshotsPolicy.arn
+}
+# END AWP proxy lambda function role
+
+# AWP proxy lambda function role policy
+# The CloudGuardAWPLambdaExecutionRolePolicy resource defines an IAM policy that is used to define the permissions for the CloudGuardAWPSnapshotsUtilsFunction.
+resource "aws_iam_policy" "CloudGuardAWPLambdaExecutionRolePolicy" {
```

Comment:
change to "CloudGuardAWPLambdaExecutionRolePolicy_InAccount" to make all resources name conssitent with the naming convention we used